### PR TITLE
refactor!: change dependencies type

### PR DESCRIPTION
**Pull request by andreapiras**
## 🖼 Context
This PR change some dependencies type to avoid code duplication consumer side

## 🚀 Changes
Change from dependency to peerDependency for "@transferwise/icons" and "@transferwise/neptune-css"

## 🤔 Considerations

This change is breaking if consumers fail to provide this dependencies 

## ✅ Checklist

- [ ] Make PR title meaningful and follow [the commit lint format](https://github.com/transferwise/neptune-web/blob/master/CONTRIBUTING.md#versioning-and-commit-lint) (it will appear in the changelog)
- [ ] Changes are tested and all tests pass
- [ ] Changes meet [accessibility standards](https://github.com/transferwise/neptune-web/blob/main/ACCESSIBILITY.md) and there are no violations in the console
- [ ] Changes work in all supported browsers (don't forget IE11)
- [ ] You've updated the documentation if necessary

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -32,6 +32,9 @@
     "@transferwise/eslint-config": "^6.0.0",
```
Comment (andreapiras) [inline package.json:32]: Ordered alphabetically and removed bootstrap dep

**Review by dozortsev — COMMENTED** [review]
please update [Setup](https://transferwise.github.io/neptune-web/branch/dependencies-reasses/about/Setup) and other related docs

**Review by dozortsev — COMMENTED** [review]
please update [bundlesize boundaries](https://bundlesize-store.now.sh/build?info=%7B%22files%22%3A%5B%7B%22maxSize%22%3A128000%2C%22path%22%3A%22.%2Fpackages%2Fcomponents%2Fbuild%2Fumd%2Fpolyfill%2Fmain.js%22%2C%22size%22%3A118671%2C%22compression%22%3A%22gzip%22%7D%2C%7B%22maxSize%22%3A110592%2C%22path%22%3A%22.%2Fpackages%2Fcomponents%2Fbuild%2Fumd%2Fno-polyfill%2Fmain.js%22%2C%22size%22%3A106050%2C%22compression%22%3A%22gzip%22%7D%2C%7B%22maxSize%22%3A10240%2C%22path%22%3A%22.%2Fpackages%2Fcomponents%2Fbuild%2Fmain.css%22%2C%22size%22%3A7271%2C%22compression%22%3A%22gzip%22%7D%2C%7B%22maxSize%22%3A44032%2C%22path%22%3A%22.%2Fpackages%2Fcss%2Fdist%2Fcss%2Fneptune.css%22%2C%22size%22%3A42039%2C%22compression%22%3A%22gzip%22%7D%5D%2C%22repo%22%3A%22transferwise%2Fneptune-web%22%2C%22branch%22%3A%22dependencies-reasses%22%2C%22commit_message%22%3A%22%22%2C%22sha%22%3A%222eb79451fc8b73aa4a88475a4eb4afc20a3e856f%22%7D)

File: bundlesize.config.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
   "files": [
     {
       "path": "./packages/components/build/umd/polyfill/main.js",
-      "maxSize": "125kB"
+      "maxSize": "120kB"
     },
     {
       "path": "./packages/components/build/umd/no-polyfill/main.js",
```
Comment (dozortsev) [inline bundlesize.config.json:8]: what about others?

Comment (andreapiras) [reply bundlesize.config.json:8]: to be honest they look pretty close so I'll leave them as they are

**Review by helendurranttw — APPROVED** [review]

File: bundlesize.config.json (hunk 2 above)
Comment (dozortsev) [reply bundlesize.config.json:8]: main.css definitely not close 🙂

**Pull request merged**